core/tx_pool.go

```diff
@@ -518,34 +518,34 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	}
 	// Ensure the transaction doesn't exceed the current block limit gas.
 	if pool.currentMaxGas < tx.Gas() {
-		return ErrGasLimit
+		return fmt.Errorf("%v currentMaxGas:%d;tx.Gas():%d", ErrGasLimit, pool.currentMaxGas, tx.Gas())
 	}
 	// Make sure the transaction is signed properly
 	from, err := types.Sender(pool.signer, tx)
 	if err != nil {
-		return ErrInvalidSender
+		return fmt.Errorf("%v err is:%v", ErrInvalidSender, err)
 	}
 	// Drop non-local transactions under our own minimal accepted gas price
 	local = local || pool.locals.contains(from) // account may be local even if the transaction arrived from the network
 	if !local && pool.gasPrice.Cmp(tx.GasPrice()) > 0 {
-		return ErrUnderpriced
+		return fmt.Errorf("%v pool.gasPrice:%d;tx.GasPrice():%d", ErrUnderpriced, pool.gasPrice, tx.GasPrice())
```

Comment:
Both pool.gasPrice and tx.GasPrice() are of big.Int type, using "%d" to print it may lead to overflow. There are two alternative ways to print it:

1. directly use **"%v"** instead of **"%d"**

2. use **"%s"** and **String** method of big.Int together